src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_gemm_emitter.cpp

```diff
@@ -70,45 +75,79 @@ void jit_gemm_emitter::validate_arguments(const std::vector<size_t>& in, const s
 
 void jit_gemm_emitter::emit_impl(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
     validate_arguments(in, out);
-    // todo: use optimized reg spill after CVS-162498
-    std::unordered_set<size_t> exclude = {};
-    store_context(exclude);
 
-    Xbyak_aarch64::XReg x0(0);
-    Xbyak_aarch64::XReg x1(1);
-    Xbyak_aarch64::XReg x2(2);
-    Xbyak_aarch64::XReg x3(3);
-    Xbyak_aarch64::XReg aux_reg(5);
-
-    // Prepare memory pointers with offsets
     std::vector<size_t> mem_ptrs_idxs{in[0], in[1], out[0]};
-    const auto& mem_ptrs = utils::transform_idxs_to_regs(mem_ptrs_idxs);
 
-    // Apply memory offsets and load adjusted pointers
-    std::vector<Xbyak_aarch64::XReg> load_regs{x1, x2, x3};
+    init_binary_call_regs(2, mem_ptrs_idxs);
+    emit_call(mem_ptrs_idxs);
+}
 
-    // Dynamically choose safe auxiliary registers that don't conflict with mem_ptrs or load_regs
-    std::vector<size_t> used_indices;
-    used_indices.reserve(mem_ptrs.size());
-    for (const auto& reg : mem_ptrs) {
-        used_indices.push_back(reg.getIdx());
-    }
-    for (const auto& reg : load_regs) {
-        used_indices.push_back(reg.getIdx());
+void jit_gemm_emitter::emit_call(const std::vector<size_t>& mem_ptrs_idxs) const {
+    const auto& call_address_reg = get_call_address_reg();
+    const auto& callee_saved_reg = get_callee_saved_reg();
+
+    std::unordered_set<size_t> exclude_spill = {};
+    store_context(exclude_spill);
+
+    auto reserved_stack_size = ov::intel_cpu::rnd_up(sizeof(GemmKaiKernelExecutor::call_args), sp_alignment);
+    emit_stack_preserve(reserved_stack_size);
+
+    const size_t A_offset = offsetof(GemmKaiKernelExecutor::call_args, A);
+    const size_t B_offset = offsetof(GemmKaiKernelExecutor::call_args, B);
+    const size_t C_offset = offsetof(GemmKaiKernelExecutor::call_args, C);
+
+    const std::vector<size_t> gemm_args_offsets = {A_offset, B_offset, C_offset};
+
+    const auto& mem_ptrs = utils::transform_idxs_to_regs(mem_ptrs_idxs);
+
+    for (size_t i = 0; i < mem_ptrs.size(); i++) {
+        const bool is_dynamic_offset = ov::snippets::utils::is_dynamic_value(m_memory_offsets[i]);
+        const bool is_valid_buffer_id = m_buffer_ids[i] != SIZE_MAX;
+
+        // Collect used register indices to avoid conflicts
+        std::vector<size_t> used_gpr_idxs = {call_address_reg.getIdx(),
+                                             callee_saved_reg.getIdx(),
+                                             mem_ptrs[i].getIdx()};
```

Comment:
I suggest you to move it before the loop `for (size_t i = 0; i < mem_ptrs.size(); i++)`, add all `mem_ptrs` to `used_gpr_idxs` and allocate `auto aux_gprs = ov::intel_cpu::aarch64::utils::get_aux_gprs(used_gpr_idxs, 3);` only once before this loop.

Because we should set all `mem_ptrs` as used gprs to avoid possible usage other `mem_ptrs` as one of `aux_gprs` during loop iteration in the current impl

Reply:
Extracted this part before the loop